File: Hands On/Week7/Day3/ContactInfo/Services/ContactService.cs
using System.Collections.Generic;
using System.Linq;
using WebApplication3.Models;

namespace WebApplication3.Services
{
    public class ContactService : IContactService
    {
        // Static list acting as in-memory database
        private static readonly List<ContactInfo> contacts =
        [
            new ContactInfo { ContactId = 1, FirstName = "John", LastName = "Doe", CompanyName = "ABC", EmailId = "john@gmail.com", MobileNo = 9876576545, Designation = "Manager" },
            new ContactInfo { ContactId = 2, FirstName = "smith", LastName = "Sharma", CompanyName = "XYZ", EmailId = "s@gmail.com", MobileNo = 9123456780, Designation = "Developer" }
        ];


        // Add new contact
        public void AddContact(ContactInfo contact)
        {
            contacts.Add(contact);
        }

        // Get all contacts
        public List<ContactInfo> GetAllContacts()
        {
            return contacts;
        }

        // Get contact by ID
        public ContactInfo GetContactById(int id)
        {
            return contacts.FirstOrDefault(c => c.ContactId == id);
        }
    }
}

File: Hands On/week11/Day51/src/app/services/contactservice.ts
import { Injectable } from '@angular/core';
import { Contact } from '../models/contact';


@Injectable({
  providedIn: 'root',
})
export class Contactservice {
  private contacts:Contact[]=[
    { id: 1, name: 'Jaya', email: 'jaya@gmail.com', phone: '9999999999' },
    { id: 2, name: 'smith', email: 'smith@gmail.com', phone: '8888888888' },
    { id: 3, name: 'Allen', email: 'allen@gmail.com', phone: '9988776655' },
    { id: 4, name: 'John', email: 'jon@gmail.com', phone: '5566778899' }
  ];

//get all contacts
getContacts():Contact[]{
  return this.contacts;
}
addContact(contact:Contact):void{
  this.contacts.push(contact);
}
getContactById(id:number):Contact|undefined{
  return this.contacts.find(c=>c.id==id);
}
}